os/fs/smartfs/smartfs_utils.c

```diff
@@ -1831,9 +2078,79 @@ int smartfs_truncatefile(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *e
  *
  ****************************************************************************/
 
+/* NOTE:- This function deviates slightly from the POSIX standard as implemented in Linux.
+ *        The file position pointer in Linux is not reset to the new data length. If a write
+ *        operation is requested after shrink, new data is written starting at the old file
+ *        position pointer. The extra space in between gets filled up with '0's.
+ *        Smartfs shrinkfile function will reset the file pointer to the position of 'length'
+ *        as requested in the shrink operation. New data is written starting at this position.
+ */
 int smartfs_shrinkfile(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs_ofile_s *sf, off_t length)
 {
-	/* TODO This will be added later */
+	int ret;
+	struct smart_read_write_s readwrite;
+	struct smartfs_chain_header_s *chainheader;
+	uint16_t sector;
+	uint16_t nextsector;
+	uint8_t data_len;
```

Comment:
```
size_t oldpos;
...
oldpos = sf->filepos;
/** TODO move above note to here **/
if (length < sf->filepos) {
    sf->filepos = length;
}
...
smartfs_seek_internal(fs, sf, length, SEEK_SET);
...
#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
...
#else
...
#endif

while() {
...
}
 /* Restore old position */
sf->filepos = oldpos;
...

Reply:
You want that new filepos should not be updated when smartfs_sync_internal is called in the middle after reducing last data sector?

Reply:
Also, I'd prefer keeping the NOTE before the start of function
It is a general description of the contrast between Linux POSIX and ours

Reply:
> You want that new filepos should not be updated when smartfs_sync_internal is called in the middle after reducing last data sector?

yes same as posix but file pos should not greater than file size.

Reply:
Ohkay